lib/srv/heartbeat.go

```diff
@@ -447,29 +447,41 @@ func (h *Heartbeat) announce() error {
 			h.setState(HeartbeatStateKeepAliveWait)
 			return nil
 		case HeartbeatModeKube:
-			kube, ok := h.current.(types.Server)
-			if !ok {
-				return trace.BadParameter("expected services.Server, got %#v", h.current)
-			}
-			keepAlive, err := h.Announcer.UpsertKubeServiceV2(h.cancelCtx, kube)
-			if err != nil {
-				// Check if the error is an Unimplemented grpc status code,
-				// if it is fall back to old keepalive method
-				// DELETE in 11.0
-				if e, ok := status.FromError(trail.ToGRPC(err)); ok && e.Code() == codes.Unimplemented {
-					err := h.Announcer.UpsertKubeService(h.cancelCtx, kube)
-					if err != nil {
-						h.nextAnnounce = h.Clock.Now().UTC().Add(h.KeepAlivePeriod)
+			var (
+				keepAlive *types.KeepAlive
+				err       error
+			)
+
+			switch current := h.current.(type) {
+			case types.Server:
```

Comment:
We're not heartbeating plain types.Server anymore, are we? Is this case branch still needed?

Reply:
We are not announcing it anymore but I had the wrong idea about how that code was called. 
Fixed